File: app/src/main/java/com/example/sunnygym/GuidedActivitiesActivity.java
package com.example.sunnygym;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.PopupMenu;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class GuidedActivitiesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_guided_activities);

        List<Activity> activities = new ArrayList<>();
        activities.add(new Activity("Zumba", R.drawable.zumba_image));
        activities.add(new Activity("Body Combat", R.drawable.body_combat_image));
        activities.add(new Activity("Body Pump", R.drawable.body_pump_image));

        // Configurar RecyclerView
        RecyclerView recyclerView = findViewById(R.id.recyclerViewActivities);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(new ActivitiesAdapter(this, activities));
    }
}

File: APP_Kotlin/app/src/main/java/com/example/sunnygym/GuidedActivitiesActivity.kt
package com.example.sunnygym

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class GuidedActivitiesActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_guided_activities)

        val activities: MutableList<Activity> = ArrayList()
        activities.add(Activity("Zumba", R.drawable.zumba_image, true))
        activities.add(Activity("Body Combat", R.drawable.body_combat_image, true))
        activities.add(Activity("Body Pump", R.drawable.body_pump_image, true))
        activities.add(Activity("Cycling", R.drawable.cycling, false))

        // Configurar RecyclerView
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerViewActivities)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = ActivitiesAdapter(this, activities)
    }
}
